### PR TITLE
Added label to system.xml

### PR DESCRIPTION
**Pull request by kyrena**
### Description

This PR add the label field to system.xml, this is an extract of #2739:

image

XML of the preview:

```xml
<head>
	<label>Crash test</label>
	<frontend_model>adminhtml/system_config_form_field_heading</frontend_model>
	<sort_order>1001</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</head>
<label_1>
	<label>label</label>
	<value>Pouet</value>
	<frontend_type>label</frontend_type>
	<frontend_model>adminhtml/system_config_form_field_label</frontend_model>
	<sort_order>1002</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</label_1>
<labelbold_2>
	<label>label bold</label>
	<value>Pouet Pouet</value>
	<bold>1</bold>
	<frontend_type>label</frontend_type>
	<frontend_model>adminhtml/system_config_form_field_label</frontend_model>
	<sort_order>1003</sort_order>
	<show_in_default>1</show_in_default>
	<show_in_website>1</show_in_website>
	<show_in_store>1</show_in_store>
</labelbold_2>
```

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

**Comment by fballiano** [comment]
Screenshot 2023-10-04 alle 14 01 21

wouldn't it be better to use the frontend_type instead of the frontend_model?

**Comment by fballiano** [comment]
also, I don't love the "store view" indicator for something that cannot change at a store view level, but I don't know if there's a way around that.

Screenshot 2023-10-04 alle 14 02 33

**Comment by kyrena** [comment]
Oh yes, true, I can remove it.

**Comment by kiatng** [comment]
How to use _Use Website_?
image

File: app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Label.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    protected function _getElementHtml(Varien_Data_Form_Element_Abstract $element)
+    {
+        $field = $element->getFieldConfig();
+
+        $label = Mage::getModel('Varien_Data_Form_Element_Label');
+        $label->setValue($field->value);
+        $label->setBold(!empty($field->bold));
+
+        return $label->getElementHtml();
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Label.php:37]: Simplify version:
```suggestion
    /**
     * @inheritdoc
     */
    public function render(Varien_Data_Form_Element_Abstract $element)
    {
        $element->unsScope()->unsCanUseWebsiteValue()->unsCanUseDefaultValue()->unsPath();
        $field = $element->getFieldConfig();
        $element->setValue($field->value);
        $element->setBold(!empty($field->bold));
        
        return parent::render($element);
```

Comment (luigifab) [reply app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Label.php:37]: Yes, but it's not the way of `Mage_Adminhtml_Block_Customer_System_Config_Validatevat` and `Mage_Adminhtml_Block_System_Config_System_Storage_Media_Synchronize `, so it's probably not the good way to simplify?

But perhaps we can add phpdoc.

Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/System/Config/Form/Field/Label.php:37]: Why is it necessary to get `$label = Mage::getModel('Varien_Data_Form_Element_Label');`?  Is it better to use the incoming `$element`?
```suggestion
        $element->setValue($field->value);
        $element->setBold(!empty($field->bold));

        return $element->getElementHtml();
```

**Comment by elidrissidev** [comment]
> wouldn't it be better to use the frontend_type instead of the frontend_model?

`<frontend_type>` alone should be enough.

**Review by luigifab — APPROVED** [review]

**Review by fballiano — CHANGES_REQUESTED** [review]
.

**Comment by fballiano** [comment]
EDIT: apologize, my comment was wrong.

the current state of the PR is this:
Screenshot 2023-10-21 alle 15 13 14

but it still requires a frontend_model, which I don't really like

**Comment by ma4nn** [comment]
Why not use the existing `Varien_Data_Form_Element_Label` or `Varien_Data_Form_Element_Note`?

**Comment by kyrena** [comment]
Because we must pass config data from `$element->getFieldConfig()`:
```php
$label->setValue($field->value);
$label->setBold(!empty($field->bold));
```

**Comment by ma4nn** [comment]
ok understood the idea now 👍  (the `Varien_Data_Form_Element_Label` can also handle the `value` but not the `bold` option)

But wouldn't it make more sense then to have a more flexible label with HTML values allowed in general (so the `bold` attribute is not necessary)? E.g. perhaps use
```xml
<frontend_type>note</frontend_type>
<frontend_model>adminhtml/system_config_form_field_label</frontend_model>
```
and then do a `$element->setText($field->value);`?

Though I agree with @fballiano that the naming is not really intuitive..

**Comment by fballiano** [comment]
IMHO the current implementation is a bit too limited:
- I think it should support full HTML as @ma4nn pointed out, removing the need for ->setBold
- using fontend_type and not requiring a frontend_model would seem more intuitive to me

**Pull request closed without merging**